q2_demux/plugin_setup.py

```diff
@@ -228,4 +230,39 @@
                  'sequences after subsampling.')
 )
 
+T = TypeMatch([SequencesWithQuality, PairedEndSequencesWithQuality,
+               JoinedSequencesWithQuality])
+plugin.methods.register_function(
+    function=q2_demux.filter_samples,
+    inputs={'samples': SampleData[T]},
+    parameters={'metadata': Metadata,
+                'where': Str,
+                'exclude_ids': Bool},
+    outputs=[
+        ('filtered_samples', SampleData[T])
+    ],
+    input_descriptions={
+        'samples': 'The demultiplexed data from which samples should be '
+        'filtered.'
+    },
+    parameter_descriptions={
+        'metadata': 'Sample metadata used with `where` parameter when '
+                    'selecting samples to retain, or with `exclude_ids` '
+                    'when selecting samples to discard.',
+        'where': 'SQLite WHERE clause specifying sample metadata criteria '
+                 'that must be met to be included in the filtered data. '
+                 'If not provided, all samples in `metadata` that are '
+                 'also in the demultiplexed data will be retained.',
+        'exclude_ids': 'If true, the samples selected by `metadata` or '
+                       '`where` parameters will be excluded from the filtered '
+                       'data instead of being retained.'
+    },
+    output_descriptions={
+        'filtered_samples': 'The samples filtered out by the metadata.'
+    },
+    name='Filter samples out of demultiplexed data.',
+    description='Filter demultiplexed data (ex. by study) based on given '
+                'metadata and optional WHERE clause.'
```

Comment:
Mention the default behavior is to retain those samples indicated (rather than remove).